File: spring/book/src/main/java/com/restful/book/model/Livro.java
package com.restful.book.model;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/// Modelo que representa um Livro.
@Data
@Document(collection = "livros")
public class Livro {

    /// Identificador único do livro.
    @Id
    private String id;

    /// Título do livro.
    @NotBlank(message = "Título é obrigatório")
    private String titulo;

    /// Autor do livro.
    @NotBlank(message = "Autor é obrigatório")
    private String autor;

    /// Descrição do livro.
    @NotBlank(message = "Descrição é obrigatória")
    private String descricao;

    /// Gênero do livro.
    @NotBlank(message = "Gênero é obrigatório")
    private String genero;

    /// URL da imagem da capa do livro.
    @NotBlank(message = "URL da imagem é obrigatória")
    private String urlImagem;
}

File: swift/book-catalogue-api-with-swift-ui-origin/Models/Livro.swift
//
//  Livro.swift
//  book-catalogue-api-with-swift-ui
//
//  Created by VINICIUS DOS SANTOS ANDRADE on 04/11/24.
//

import Foundation

/// Modelo que representa um Livro.
struct Livro: Codable, Identifiable {
    /// Identificador único do livro.
    var id: UUID = UUID()
    /// Título do livro.
    var titulo: String
    /// Autor do livro.
    var autor: String
    /// Descrição do livro.
    var descricao: String
    /// Gênero do livro.
    var genero: String
    /// URL da imagem da capa do livro.
    var urlImagem: String
}
